### PR TITLE
[Issue 14073] Fixes key shared subscription: hash ranges are not fully reported in stats.

### LINKED ISSUE #14073: Key shared subscription: hash ranges are not fully reported in stats
**Describe the bug**
When I run `pulsar-admin topics stats TOPIC-NAME` and I have a KeyShared subscription with stickyHashRanges, with many ranges, the stats are not fully reporting the ranges, it looks like that for each consumer they report only the last set of ranges assigned to each Consumer and for some Consumers the list is empty even if the Consumer set the range and it is actually consuming messages.

There should be some bug here:
https://github.com/apache/pulsar/blob/9abd6d30f39f74e255cd5dac40b77af3dce5c468/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/HashRangeExclusiveStickyKeyConsumerSelector.java#L69

  

```
 public Map<Consumer, List<Range>> getConsumerKeyHashRanges() {
        Map<Consumer, List<Range>> result = new HashMap<>();
        Map.Entry<Integer, Consumer> prev = null;
        for (Map.Entry<Integer, Consumer> entry: rangeMap.entrySet()) {
            if (prev == null) {
                prev = entry;
            } else {
                if (prev.getValue().equals(entry.getValue())) {
                    result.computeIfAbsent(entry.getValue(), key -> new ArrayList<>())
                            .add(Range.of(prev.getKey(), entry.getKey()));
                }
                prev = null;
            }
        }
        return result;
    }
```

**Additional context**
Pulsar 2.8.2

### PR DESCRIPTION
**Pull request by mattisonchao**
Fixes #14073

Master Issue: #14073

### Motivation

see #14073

### Modifications

- Fixes ``getConsumerKeyHashRanges`` logic.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

### Documentation

  
- [x] `no-need-doc`

### CODE REVIEW

**Comment by eolivelli** [comment]
This patch is not bad, it is only missing tests

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/HashRangeExclusiveStickyKeyConsumerSelector.java [hunk 1]
```diff
@@ -67,20 +68,22 @@ public void removeConsumer(Consumer consumer) {
...
-        return result;
+        return rangeMap.entrySet().stream()
+                .collect(Collectors.groupingBy(Map.Entry::getValue))
+                .entrySet().stream().map(groupedEntry -> {
+                    Consumer consumer = groupedEntry.getKey();
+                    List<Integer> consumerHash = groupedEntry.getValue().stream()
+                            .map(Map.Entry::getKey)
+                            .sorted(Integer::compareTo)
+                            .collect(Collectors.toList());
+                    List<Range> ranges = Lists.newArrayList();
```
Comment (eolivelli) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/HashRangeExclusiveStickyKeyConsumerSelector.java:79]: IIRC we stopped using Lists recently

Comment (mattisonchao) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/HashRangeExclusiveStickyKeyConsumerSelector.java:79]: sure, i will fix it.

**Pull request closed without merging**